<a href="https://colab.research.google.com/github/Jelladotexe/stat131/blob/main/stat-mid01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# แบบฝึกหัด 1

---

### 1.1

***เขียนโปรแกรมเพื่อเปลี่ยนค่าจากองศาเซลเซียสไปเป็นองศาฟาเรนไฮต์และเคลวิน ดังนี้***

F = (9/5)C + 32  
K = C + 273.15

รับข้อมูลองศาเซลเซียส

In [131]:
C = [39.85, 25, 32]

###  
แปลงเป็นองศาฟาเรนไฮต์ให้ได้ผลลัพธ์เป็น list เก็บใน F

In [132]:
F = [i*(9/5)+32 for i in C]
F

[103.73, 77.0, 89.6]

ผลลัพธ์ที่ต้องการแสดง [103.73, 77.0, 89.6]

###  
แปลงเป็นองศาฟาเคลวินให้ได้ผลลัพธ์เป็น list เก็บใน K

In [133]:
K = [i+273.15 for i in C]
K

[313.0, 298.15, 305.15]

ผลลัพธ์ที่ต้องการแสดง [313.0, 298.15, 305.15]

###  
แสดงการสรุปผล

In [134]:
for i in range(3):
  print(f"{C[i]}°C = {F[i]} °F = {K[i]} °K")

39.85°C = 103.73 °F = 313.0 °K
25°C = 77.0 °F = 298.15 °K
32°C = 89.6 °F = 305.15 °K


ผลที่ต้องการแสดง  
39.85 °C = 103.73 °F = 313.0 °K  
25 °C = 77.0 °F = 298.15 °K  
32 °C = 89.6 °F = 305.15 °K  

---

### 1.2

ให้นิสิตแสดงวิธีการแปลง list income (ซึ่งเปรียบเสมือนเป็นข้อมูลเงินเดือนของผู้ตอบแบบสอบถาม 20 ท่าน)  
ให้เป็น numpy array เก็บในตัวแปรที่ชื่อ income_number โดย array ที่แปลงได้ควรมีชนิด dtype เป็น float   
การแปลง string ให้เป็น float กำหนดให้ K = 1000 และ M = 1000000   
string "-" ให้แปลงเป็น np.nan (dtype ต้องเป็น float จึงจะใช้ np.nan ได้)

\* โจทย์ข้อนี้สามารถทำได้หลายวิธี  
หากใช้ numpy นิสิตอาจต้องหาข้อมูลเพิ่ม เช่น การใช้ np.char.replace หรือฟังก์ชัน eval

In [135]:
income = ["20K", "3M", "12K", "0.04M", "0.023K", "-", "16K", "740K", "15K", "40K",
         "-", "17K", "-", "120K", "56K", "-", "42K", "2M", "0.14M", "60K"]


In [136]:
import numpy as np

In [137]:
income_num = np.array(income)
income_num = np.char.replace(income_num,'K','*1000')
income_num = np.char.replace(income_num,'M','*1000000')
income_num = np.char.replace(income_num,'-','0')
income_num = np.array([eval(i) for i in income_num])
income_num = np.where(income_num == 0.0e+00,np.nan,income_num)
income_num

array([2.0e+04, 3.0e+06, 1.2e+04, 4.0e+04, 2.3e+01,     nan, 1.6e+04,
       7.4e+05, 1.5e+04, 4.0e+04,     nan, 1.7e+04,     nan, 1.2e+05,
       5.6e+04,     nan, 4.2e+04, 2.0e+06, 1.4e+05, 6.0e+04])

##  
วิธีการของนิสิตควรได้ผลลัพธ์ดังที่ระบุเมื่อรัน cell ต่อไปนี้

\*\*\* ต้องขออภัยผลที่ระบุจะเป็นการรัน cell โดยไม่ใช้คำสั่ง print (ความจริงแล้วหากไม่ระบุคำสั่ง print จะเป็นการเรียกคำสั่ง display)

ผลลัพธ์ที่ควรแสดง: array([2.0e+04, 3.0e+06, 1.2e+04, 4.0e+04, 2.3e+01,     nan, 1.6e+04,
       7.4e+05, 1.5e+04, 4.0e+04,     nan, 1.7e+04,     nan, 1.2e+05,
       5.6e+04,     nan, 4.2e+04, 2.0e+06, 1.4e+05, 6.0e+04])

In [138]:
type(income)

list

ผลลัพธ์ที่ควรแสดง: numpy.ndarray

In [139]:
income_num.dtype

dtype('float64')

ผลลัพธ์ที่ควรแสดง: dtype('float64')

In [140]:
np.nanmean(income_num)

np.float64(394876.4375)

ผลลัพธ์ที่ควรแสดง: 394876.4375

In [141]:
np.nanmedian(income_num)

np.float64(41000.0)

ผลลัพธ์ที่ควรแสดง: 41000.0